### PR TITLE
Check filefino extention is loaded while uploading

### PR DESCRIPTION
**Pull request by mszb**
Added function to check either fileinfo extention is loaded or not.
Before this check, if extention was not loaded, script jsut stop and
there is no log of what happend. Now this will generate an error log and
return.

### CODE REVIEW

File: system/libraries/Upload.php [hunk 1]
```diff
@@ -1218,6 +1218,13 @@ protected function _file_mime_type($file)
 		// We'll need this to validate the MIME info string (e.g. text/plain; charset=us-ascii)
 		$regexp = '/^([a-z\-]+\/[a-z0-9\-\.\+]+)(;\s.+)?$/';
 
+		//Check fileinfo extention is loaded or not
+		if(!extension_loaded('fileinfo'))
```
Comment (gxgpet) [inline system/libraries/Upload.php]: A space [should](http://www.codeigniter.com/user_guide/general/styleguide.html#logical-operators) always precede and follow `!`.

**Comment by narfbg** [comment]
The extension in question is [enabled by default since PHP 5.3](https://secure.php.net/manual/en/fileinfo.installation.php), which is why a `function_exists()` check was removed in a838279625becfba98ccb7635d35c67297129c42.

If we're to do anything about that, it would be to revert that change.

Referencing #4887.

**Comment by andersonunsonst** [comment]
but, windows users yet need enable this, it would be interesting a alert with throw exception

**Comment by mszb** [comment]
@narfbg .. yes i agree with @andersonunsonst . I've counter this issue in windows systems. That's why i created this pull-request. Because, if file info extension is not enable and there is no error log, user have no clue whats going on.

**Comment by narfbg** [comment]
*sigh*

My first sentence explains *how and why* this issue appeared. And the reference to #4887 was meant to point you to an *open* issue where we're investigating it further.

And when I said this:

> If we're to do anything about that, it would be to revert that change.

I meant that if any change should be made, it would be to make it like [this](https://github.com/bcit-ci/CodeIgniter/blob/3.1.0/system/libraries/Upload.php#L1215) again.

**Pull request closed without merging**